lib/middleware.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+// Chain is essentially a linked list of middlewares to run in order.
+function Chain() {
+  var self = this;
+  var middlewares = [];
+
+  this.getMiddlewares = function() {
+    return middlewares;
+  };
+
+  this.add = function(middleware) {
+    if (typeof middleware !== 'function')
+      throw new Error('attempted to add non-function middleware');
+    if (self.contains(middleware))
```

Comment:
Any reason this use this vs `middlewares.indexOf(middleware) !== -1`?

Reply:
I was going to cite browser compatibility reasons (preferring to be more broad) but we DO use indexOf in several other spots. Will switch for conciseness.